src/pages/studyView/charts/ChartContainer.tsx

```diff
@@ -198,17 +198,21 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
 
     readonly survivalChartData = remoteData({
         // patientToAnalysisGroup assumed defined, since we're calling survivalChartData
-        await:()=>[this.props.promise, this.props.patientToAnalysisGroup!],
-        invoke:()=>{
-            return Promise.resolve(
-                makeSurvivalChartData(
-                    this.props.promise.result!.alteredGroup.concat(this.props.promise.result!.unalteredGroup),
-                    this.props.analysisGroupsSettings.groups,
-                    this.props.patientToAnalysisGroup!.result!,
-                    this.naPatientsHiddenInSurvival,
-                    this.props.patientKeysWithNAInSelectedClinicalData,
-                )
-            );
+        await: () => [this.props.promise, this.props.patientToAnalysisGroup!],
+        invoke: () => {
+            if (this.props.promise.result === undefined) {
+                return Promise.resolve(undefined);
```

Comment:
you could directly return undefined or object instead of creating a promise using async. something like
```
invoke: async () => {
            if (this.props.promise.result !== undefined) {
                return makeSurvivalChartData(...)
            }
            return undefined;
        }
```